src/ducks/appSuggestions/services.spec.js

```diff
@@ -8,28 +7,37 @@ describe('findSuggestionForTransaction', () => {
   const transaction = { _id: 'o1', label: 'boulanger' }
 
   it('should return a fresh io.cozy.apps.suggestions document', async () => {
-    jest.spyOn(AppSuggestion, 'fetchBySlug').mockResolvedValueOnce(null)
-
-    const suggestion = await findSuggestionForTransaction(transaction, brands)
+    const suggestions = []
+    const suggestion = await findSuggestionForTransaction(
+      transaction,
+      brands,
+      suggestions
+    )
 
     expect(suggestion).toMatchSnapshot()
   })
 
   it('should update an existing io.cozy.apps.suggestions document', async () => {
-    jest.spyOn(AppSuggestion, 'fetchBySlug').mockResolvedValueOnce({
-      slug: 'boulanger',
-      silenced: false,
-      reason: {
-        code: 'FOUND_TRANSACTION'
-      },
-      relationships: {
-        transactions: {
-          data: [{ _id: 'o2', _type: TRANSACTION_DOCTYPE }]
+    const suggestions = [
+      {
+        slug: 'boulanger',
+        silenced: false,
+        reason: {
+          code: 'FOUND_TRANSACTION'
+        },
+        relationships: {
+          transactions: {
+            data: [{ _id: 'o2', _type: TRANSACTION_DOCTYPE }]
+          }
         }
       }
-    })
+    ]
 
-    const suggestion = await findSuggestionForTransaction(transaction, brands)
+    const suggestion = await findSuggestionForTransaction(
+      transaction,
+      brands,
+      suggestions
+    )
```

Comment:
I could not see a test that was testing that we are using the originalSuggestion if it exists ? I think we need one if it does not exist.

Reply:
That's what this one is testing: we have an existing suggestion, and `findSuggestionForTransaction` uses it and updates it to add the transaction to the suggestion's relationships.